### ISSUE TITLE
Duration.fromISOTime('0:00') --> can't be parsed

### ISSUE THREAD

**Issue by carstenfuchs** [moment/luxon#1460]
**Describe the bug**
The attempt to create a Duration with `fromISOTime()` fails whenever the hours are not two digits.

(As a side note: the attempt to create a Duration from time 50:00 works. See example below.)

**To Reproduce**
These are all invalid:
```js
luxon.Duration.fromISOTime("0:00")
luxon.Duration.fromISOTime("1:00")
luxon.Duration.fromISOTime("222:00")
```

Observe that in contrast, this *is* valid:
```js
luxon.Duration.fromISOTime("50:00")
```

**Actual vs Expected behavior**
All of these should result in valid Durations:
```js
luxon.Duration.fromISOTime("0:00")
luxon.Duration.fromISOTime("1:00")
luxon.Duration.fromISOTime("222:00")
```

(Alternatively, "50:00" should be invalid.)

**Desktop (please complete the following information):**

- OS: Ubuntu 22.04 LTS
- Browser: Firefox 115.0
- Luxon version: 3.3.0
- Your timezone: "Europe/Berlin"

**Additional context**
In my opinion, the root cause of this problem is that *times* and *durations* are not actually compatible:

While a time such as 25:00 does not make sense (and is not covered by ISO 8601), a duration of 25 hours does.
Specifying durations with hours, minutes and seconds in the form HH:MM:SS.s also makes sense, but should not be limited to the limits of times given in the same form.
(Strictly spoken, the method name `fromISOTime()` is an antagonism: it should possibly be named `fromHours()` or similarly.)

In summary, my proposal is to relax `Duration.fromISOTime()` to accept all of the examples above, that is, hours that have one, two or more digits.

**Comment by diesieben07** [comment]
ISO refers to ISO 8601, which requires leading zeros.
However you are correct in that `fromISOTime` should not technically not accept invalid times of day (such as your example 50:00). It is debatable whether we should make `fromISOTime` produce an invalid Duration for invalid times of day and if that is worth the breaking change.
ISO 8601 specifies a different format for actual durations (unrelated to time of day), which can be parsed by `Duration.fromISO`.

**Comment by carstenfuchs** [comment]
Yes, I think that there are two “clean” solutions for this:

  1. Either strictly apply ISO 8601, which would cause times such as 50:00 (that are valid today) to become invalid.
  2. Further relax (or, in a sense, forego) the ISO 8601 rules so that examples such as 0:00 or 123:45 become valid.

It would be ideal if there was a standards or normative reference for durations that are given in this form, but I've not be able to find any (maybe DIN 5008 has more than I have been able to dig up?) But even without such a normative reference, I think that the 2. option, relaxing the rules, would be the right choice for the following reasons:

  - For durations a notation can be used that looks the same or very much like the notation that is used for times, e. g. 11:23 is a valid example for both, but durations and times are conceptually entirely different subjects. Personally, I think that it is self-suggesting to use time formatting also for formatting durations, but it is generally rather by accident than by intention that time formatting limitations are also forced upon durations.
  - One characteristic of times and durations being different is that durations can be added, but times cannot: Adding durations 12:00 and 13:00 yields a valid duration, but adding times 12:00 and 13:00 is undefined.
  - Well … sport events such as soccer matches that have a length of 1 hour and 30 minutes are given as 90:00 … Yes, this is besides the point. But I think that enough real world examples exist where durations of more than 24 hours are given in HHH:MM form.
